docs/design/unsupported-args-design.md

```diff
@@ -0,0 +1,118 @@
+
+## Non Goals
+- Deprecate current ways of defining extra arguments.
+- Change precedences within current ways of defining extra arguments.
+- Validation of unsupported args, they are passed to the appropriate executables without any prior validation.
+
+## Design
+Unsupported arguments are defined as a `ConfigMap` within the same Namespace as OADP Operator.
+
+There may be multiple `ConfigMaps` for particular section of the OADP configuration.
+
+There are default naming convention for the `ConfigMaps`, however user may override those using `DPA` annotations.
+
+When ConfigMap is found, controller takes those values as the highest priority and replaces **all** of the other configuration options, even if they were defined within `DPA` schema for the section for which the `ConfigMap` was created.
+
+The Schema of an `ConfigMap` includes data that corresponds to the arguments passed to the relevant executable as in the below example:
+
+   ```
+   apiVersion: v1
+   kind: ConfigMap
+   metadata:
+     name: oadp-unsupported-velero-server-args
+     namespace: openshift-adp
+   data:
+     --default-volume-snapshot-locations: 'aws:backups-primary,azure:backups-secondary'
+     --log-level: 'debug'
+   ```
+
+If the name of the ConfigMap is recognized by the OADP controller it is consumed, otherwise user has to create DPA with the annotation pointing to a relevant ConfigMap as in the example:
+
+
+    ```yaml
+    kind: DataProtectionApplication
+    apiVersion: oadp.openshift.io/v1alpha1
+    metadata:
+    name: sample-dpa
+    namespace: openshift-adp
+    annotations:
+        custom.configmap.dpa/velero-server-args: 'oadp-unsupported-velero-server-args'
+    spec:
+        [...]
+    ```
+
+## Alternatives Considered
+ - Resource specific configuration / CRD options
+ 
+   The configuration data is directly integrated into the Custom Resource Definition (CRD). This is how currently some of the extra arguments are defined within OADP.
+
+   There is however schema enforcement, which is problematic, because it requires new release of OADP to include unknown or not available options.
+   
+   This approach also introduces more complexity as it requires logic to handle and validate each of the options. On top of that having CRD options introduces migration and upgrade problems requiring migration strategies to be taken into account for existing resources. 
+
+   Adding the unsupported arguments and options within CRD makes those options highly visible to the users and may encourage to use them.
+
+ - Annotations to include unsupported args directly
+   This mmethod allows easily to add args to existing resources, however they have size limitations, restricting the amount of data that can be stored and may have impact on the scenarios where limit is reached.
+
+   Annotations are also problematic to maintain when used for more complex types of configurations.
+
+   Sample custom unsupported args:
+
+    ```yaml
```

Comment:
rendering did not work here 
![Screenshot from 2024-05-29 13-19-41](https://github.com/openshift/oadp-operator/assets/66965232/8491b9c5-46db-4a0a-9793-aa9aee0ecd34)

Reply:
wired, it's perfectly fine on my rendering:

![image](https://github.com/openshift/oadp-operator/assets/16115240/51a6fb8c-0ba9-42e3-826c-36a17de27fd2)

Reply:
sorry, put comment in wrong place. Its the second code block (beginning in line 49) that is not rendering. I believe its because it has 2 indents instead of 1 (or 4 spaces instead of 2)

Reply:
ah see it